File: 824-number-of-lines-to-write-string/number-of-lines-to-write-string.java
class Solution {
    public int[] numberOfLines(int[] widths, String s) {
        int r=1,c=0;
        for(char ch:s.toCharArray()){
            int width=widths[ch-'a'];
            r=width+c>100?r+1:r;
            c=width+c>100?width:width+c;
        }
        return new int[]{r,c};
    }
}

File: 824-number-of-lines-to-write-string/number-of-lines-to-write-string.py
class Solution(object):
    def numberOfLines(self, widths, s):
        """
        :type widths: List[int]
        :type s: str
        :rtype: List[int]
        """
        r=1
        c=0
        for i in s:
            width=widths[ord(i)-ord('a')]
            r+=1 if width+c >100 else 0
            c=width if width+c >100 else c+width
        return [r,c]